mqtt/owner_consent_agent_client .go

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+
+package mqtt
+
+import (
+	"fmt"
+
+	"github.com/eclipse-kanto/update-manager/api"
+	"github.com/eclipse-kanto/update-manager/api/types"
+	"github.com/eclipse-kanto/update-manager/logger"
+
+	pahomqtt "github.com/eclipse/paho.mqtt.golang"
+	"github.com/pkg/errors"
+)
+
+type ownerConsentAgentClient struct {
+	*mqttClient
+	domain  string
+	handler api.OwnerConsentAgentHandler
+}
+
+// NewOwnerConsentAgentClient instantiates a new client for triggering MQTT requests.
+func NewOwnerConsentAgentClient(domain string, config *ConnectionConfig) (api.OwnerConsentAgentClient, error) {
+	client := &ownerConsentAgentClient{
+		mqttClient: newInternalClient(domain, newInternalConnectionConfig(config), nil),
+		domain:     domain,
+	}
+	pahoClient, err := newClient(client.mqttConfig, client.onConnect)
+	if err == nil {
+		client.pahoClient = pahoClient
+	}
+	return client, err
+}
+
+func (client *ownerConsentAgentClient) onConnect(_ pahomqtt.Client) {
```

Comment:
If we can't subscribe for owner consent does it make sense to try to stop or restart the service? Otherwise just with error messages this issue could be masked.

Reply:
As the connection is to the local MQTT broker, it is not likely to fail. I am not fan of repeating what has just failed. As this client behaves like a library, we either need to stick to just logging or introduce some callback to export the error.